packages/ui/src/modal/buy/BuyModalRenderer.tsx

```diff
@@ -689,7 +691,7 @@ export const BuyModalRenderer: FC<Props> = ({
       setTransactionError(null)
       setStepData(null)
       setSteps(null)
-      setQuantity(1)
+      setQuantity(defaultQuantity || 1)
```

Comment:
We should be setting this when the modal opens, not when the modal closes (so we can get the latest)